lib/Telemetry/Monitor.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @author Thomas Müller <thomas.mueller@tmit.eu>
+ *
+ * @copyright Copyright (c) 2024, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\ConfigReport\Telemetry;
+
+require_once __DIR__ . '/../../vendor/autoload.php';
+
+use OC\License\LicenseFetcher;
+use OCA\ConfigReport\ReportDataCollector;
+use OCP\Http\Client\IClientService;
+use OCP\IConfig;
+use OCP\ILogger;
+use Ramsey\Uuid\Uuid;
+
+class Monitor {
+    /** @var IConfig */
+    private $config;
+    /** @var ILogger */
+    private $logger;
+    /** @var IClientService */
+    private $clientService;
+    /** @var ReportDataCollector */
+    private $collector;
+
+    public function __construct(IConfig $config, ILogger $logger, IClientService $clientService, ReportDataCollector $collector) {
+        $this->config = $config;
+        $this->logger = $logger;
+        $this->clientService = $clientService;
+        $this->collector = $collector;
+    }
+
+    public function sendTelemetry(): bool {
+        # only customer telemetry is sent over
+        $licenseKey = $this->getLicenseKey();
+        if ($licenseKey === "") {
+            return false;
+        }
+
+        # telemetry ia an opt-out feature
+        $telemetryOptIn = $this->config->getSystemValue('telemetry.enabled', true);
```

Comment:
```suggestion
        # telemetry is an opt-in feature
        $telemetryOptIn = $this->config->getSystemValue('telemetry.enabled', false);
```

-> https://github.com/owncloud/configreport/pull/197#discussion_r1667086676